File: Python/ParkingLot.py
# import class Vehicle dari file Vehicle
from Vehicle import Vehicle

# deklarasi kelas

class ParkingLot (Vehicle):

    # atribut private
    __kapasitas = ""
    __jml_kendaraan = ""
    # composite kelas Garage
    __Garage = []  # array of object Garage

   # konstruktor
    def __init__(self, kapasitas="", jml_kendaraan="",  Garage=[]):
        self.__kapasitas = kapasitas
        self.__jml_kendaraan = jml_kendaraan
        self.__Garage = Garage

    # Getter and Setter
    def get_kapasitas(self):
        return self.__kapasitas

    def set_kapasitas(self, kapasitas):
        self.__kapasitas = kapasitas

    def get_jml_kendaraan(self):
        return self.__jml_kendaraan

    def set_jml_kendaraan(self, jml_kendaraan):
        self.__jml_kendaraan = jml_kendaraan

    def get_Garage(self):
        return self.__Garage

    def set_Garage(self, Garage):
        self.__Garage = Garage

File: CPP/ParkingLot.cpp
// import file header
#include "header.hh"

/* konstruktor */

// konstruktor tanpa parameter
ParkingLot::ParkingLot()
{
    // set isi atribut secara default

    this->kapasitas = "";
    this->jumlah_kendaraan = "";
}

// konstruktor dengan parameter
ParkingLot::ParkingLot(string kapasitas, string jumlah_kendaraan, Garage garasi)
{
    // set isi atribut berdasarkan isi parameter
    this->kapasitas = kapasitas;
    this->jumlah_kendaraan = jumlah_kendaraan;
    this->garasi.push_back(garasi);
}

/* Getter dan Setter*/

// Get kapasitas
string ParkingLot::get_kapasitas()
{
    return this->kapasitas;
}

// Set kapasitas
void ParkingLot::set_kapasitas(string kapasitas)
{
    this->kapasitas = kapasitas;
}
// Get jumlah_kendaraan
string ParkingLot::get_jumlah_kendaraan()
{
    return this->jumlah_kendaraan;
}

// Set jumlah_kendaraan
void ParkingLot::set_jumlah_kendaraan(string jumlah_kendaraan)
{
    this->jumlah_kendaraan = jumlah_kendaraan;
}

// Get MataKuliah
list<Garage> ParkingLot::get_garasi()
{
    return this->garasi;
}

// Set MataKuliah
void ParkingLot::set_garasi(list<Garage> garasi)
{
    this->garasi = garasi;
}
// add list objek MataKuliah
void ParkingLot::add_garasi(Garage garasi)
{
    this->garasi.push_back(garasi);
}

/* desktruktor */

// default
ParkingLot::~ParkingLot() // desktruktor
{
}
